### PR TITLE
configury: patch configure in order to correctly support flang compilers

### PR DESCRIPTION
**Pull request by ggouaillardet**
Signed-off-by: Gilles Gouaillardet <gilles@rist.or.jp>

### CODE REVIEW

**Comment by ggouaillardet** [comment]
@jsquyres could you please review this ?

currently, Open MPI cannot be built out of the box with flang or (flang based) ARM compiler.
as described at https://developer.arm.com/products/software-development-tools/hpc/resources/porting-and-tuning/building-openmpi-with-arm-compiler a few extra steps are needed.
the root cause is libtool does not support flang compilers, but since we already hack configure in order to get support for what we need, i just made yet an other hack.

@PHHargrove by any chance, would you be able to test this PR with ARM compilers (i only tested flang on linux/x86_64) ? note you need recent autotools and run ```autogen.pl```

**Comment by PHHargrove** [comment]
@ggouaillardet I don't have flang access on ARM64.  
I don't even have gcc-6 or gcc-7!    
I think @shamisp is likely a better resource for ARM64 testing.

**Comment by ggouaillardet** [comment]
thanks @PHHargrove 

@shamisp could you please give this PR a try and provide some feedback ?

File: autogen.pl [hunk 1]
```diff
@@ -979,6 +979,26 @@ sub patch_autotools_output {
 
         # We have to change the search pattern and substitution on each
         # iteration to take into account the tag changing
+        my $search_string = '# Lahey Fortran 8.1.\n.*\n\s+' .
```
Comment (jsquyres) [inline autogen.pl]: Why is Lahey Fortran mentioned again down around new line 1026?

Comment (ggouaillardet) [inline autogen.pl]: thanks, i will double check that tomorrow.
(new) line 1026 has a different purpose, but my ```$search_string``` might match more expressions than needed

Comment (ggouaillardet) [inline autogen.pl]: i checked and each ```$search_string``` matches a different expression.
fwiw, the new ```$search_string``` is used to insert ```flang``` related stuff, whereas the previous one is used to enhance Lahey Fortran support.~

**Comment by ggouaillardet** [comment]
@jsquyres i made the changes. latest libtool does not support ```flang``` so i submitted a patch upstream. meanwhile, we can hack the generated ```configure``` in ```autogen.pl``` as we do for similar purposes

**Comment by shamisp** [comment]
@pawosm-arm can you please take a look at the flang patch ?

File: autogen.pl [hunk 2]
```diff
@@ -979,6 +979,28 @@ sub patch_autotools_output {
 
         # We have to change the search pattern and substitution on each
         # iteration to take into account the tag changing
+        my $search_string = '# icc used to be incompatible with GCC.\n\s+' .
```
Comment (pawosm-arm) [inline autogen.pl]: did we really have to introduce one more foreach loop? Couldn't the one that looks for Sun's compiler be reused?

File: autogen.pl [hunk 3]
```diff
@@ -998,7 +1020,6 @@ sub patch_autotools_output {
         push(@verbose_out, $indent_str . "Patching configure for Sun Studio Fortran version strings ($tag)\n");
         $c =~ s/$search_string/$replace_string/;
     }
-
```
Comment (pawosm-arm) [inline autogen.pl]: Don't add or remove lines.

File: autogen.pl (hunk 2 above)
Comment (ggouaillardet) [reply autogen.pl]: this is the way we have always done. it makes it easier to follow the logic of what we are doing.

an empty line was removed by mistake, and i fixed that

**Pull request merged**